### PR TITLE
Simplify crossgen script and move to dedicated target in redist.csproj

### PR DESCRIPTION
**Pull request by piotrpMSFT**
/cc @livarcocc @jgoshi

### CODE REVIEW

File: src/redist/redist.csproj [hunk 1]
```diff
@@ -48,29 +48,37 @@
...
-    <WriteLinesToFile File="$(SdkOutputDirectory)/.version"
+    <WriteLinesToFile File="$(PublishDir)/.version"
                       Lines="$(CommitHash);$(SdkVersion);$(Rid)"
                       Overwrite="true" />
   </Target>
 
+  <Target Name="PublishRoslyn"
+          BeforeTargets="Publish">
+          
+    <DotNetPublish ToolPath="$(Stage0Directory)"
```
Comment (livarcocc) [inline src/redist/redist.csproj]: Should this be here now? Or should this just be a dependency between redist and tool_roslyn?

**Comment by livarcocc** [comment]
Make sure that the chmod we run on the files is not being overwritten by crossgen.

**Comment by piotrpMSFT** [comment]
@livarcocc chmod still runs after this publish step finishes. I believe next on my docket will be moving chmod out into it's own target. I'll work on speeding it up as well... we currently spend ~30s chmod'ing.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Debian8.2 x64 Debug Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Windows_NT x64 Release Build please.

**Comment by piotrpMSFT** [comment]
@mmitche it looks like our Windows x64 queue is REALLY backed up. Can we spin up some additional machines?

**Comment by mmitche** [comment]
@piotrpMSFT More are coming online as as Jenkins spins up more machines.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Ubuntu x64 Release Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Ubuntu x64 Release Build please.

**Pull request merged**